azkaban-common/src/main/java/azkaban/executor/container/ContainerizedExecutionManager.java

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.executor.container;
+
+import azkaban.Constants;
+import azkaban.Constants.ContainerizedExecutionManagerProperties;
+import azkaban.DispatchMethod;
+import azkaban.executor.AbstractExecutorManagerAdapter;
+import azkaban.executor.ExecutableFlow;
+import azkaban.executor.Executor;
+import azkaban.executor.ExecutorApiGateway;
+import azkaban.executor.ExecutorLoader;
+import azkaban.executor.ExecutorManagerException;
+import azkaban.executor.Status;
+import azkaban.metrics.CommonMetrics;
+import azkaban.utils.FileIOUtils.LogData;
+import azkaban.utils.Pair;
+import azkaban.utils.Props;
+import java.io.IOException;
+import java.lang.Thread.State;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class ContainerizedExecutionManager extends AbstractExecutorManagerAdapter {
```

Comment:
I feel the naming of this class is somewhat confusing. Since the class is about Dispatch, in my opinion it should include that in the naming. Also, please add some comments on top of the class describing its purpose.

Reply:
Took care of it. Thank you.